modin/numpy/arr.py

```diff
@@ -734,6 +734,16 @@ def __abs__(
 
     absolute = __abs__
 
+    def __iter__(self):
+        """
+        Return an iterator of the values.
+
+        Returns
+        -------
+        iterable
+        """
+        return self._to_numpy().__iter__()
```

Comment:
@RehanSD I can implement in this way. This is more flexible, and there is no need to transfer the implementation of `_to_numpy` function to the query compiler. What do you think?

```suggestion
        if hasattr(self._query_compiler, "numpy__iter__"):
            return self._query_compiler.numpy__iter__()
        return self._to_numpy().__iter__()
```

Reply:
I personally do not like the idea of such a dynamic interface... I would rather this `numpy__iter__()` would do this `._to_numpy().__iter__()` thing itself...

That said, why are we even wanting to default to pandas (or numpy) in such a simple case? Maybe just implement the iterator ourselves without even defaulting?

Reply:
It probably makes sense to implement an iterator, but right now I don't have the ability to do it.